unattended-upgrade

```diff
@@ -1637,13 +1693,21 @@ def main(options, rootdir=""):
         #
         # this step may also remove _auto-removable_ reverse dependencies
         # of kernel packages
-        auto_removable_kernel_pkgs = only_kernel_pkgs(auto_removable)
+        auto_removable_kernel_pkgs = {
```

Comment:
I would love to see some unit tests around this new feature but I do understand this is a bit tricky.